.github/workflows/bot.yml

```diff
@@ -183,3 +187,42 @@ jobs:
           key: criterion_benchmarks_${{ github.sha }}
           restore-keys: |
             criterion_benchmarks_
+
+  Criterion_compare:
+    name: Comparing Micro Benchmarks
+    if: contains(github.event.pull_request.labels.*.name, 'benchmark') || github.event_name == 'push'
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+      contents: write
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+
+      - name: Restore file
+        uses: actions/cache@v2
+        with:
+          path: benches/main_benchmarks.json
+          key: criterion_benchmarks_${{ github.sha }}
+          restore-keys: |
+            criterion_benchmarks_
+
+      - name: Print Criterion Benchmarks
+        run: |
+          cargo install cargo-criterion rust-script
+          cargo criterion --message-format=json > benches/benchmarks.json
+          rust-script benches/parse_json_benchmark benches/benchmarks.json benches/change_benchmarks.md
+          cat benches/change_benchmarks.md
```

Comment:
- Keep the `cat` step to "Print Benchmarks"
- Create a new "Run Benchmarks" step that performs the running of the benchmarks and creation of the md file.